File: sources/Physics/Orbits.py
from numba import njit
from numpy import array, sqrt 


#@jit(nopython=True)
@njit
def Kepler(U, t): 

    x = U[0]; y = U[1]; dxdt = U[2]; dydt = U[3]
    d = ( x**2  +y**2 )**1.5

    return  array( [ dxdt, dydt, -x/d, -y/d ] ) 

@njit
def Arenstorf_equations(U, t):          
   
     mu = 0.012277471
           
     x = U[0]; y = U[1]; vx = U[2];  vy = U[3]   
 
     D1 = sqrt( (x+mu)**2     + y**2 )**3
     D2 = sqrt( (x-(1-mu))**2 + y**2 )**3
     
     dxdt = vx  
     dydt = vy  
     dvxdt = x + 2 * vy - (1-mu)*( x + mu )/D1 - mu*(x-(1-mu))/D2
     dvydt = y - 2 * vx - (1-mu) * y/D1 - mu * y/D2
    
     return array( [ dxdt, dydt, dvxdt, dvydt ] ) 

File: sources/milestones (julia)/Physics/Orbits.jl
module Orbits 

 function Kepler(U, t)

    # x, y, dxdt, dydt = U[1], U[2], U[3], U[4]
    x = U[1]; y =U[2];  dxdt = U[3]; dydt = U[4]
    d = ( x^2 + y^2 )^1.5

    return   [ dxdt, dydt, -x/d, -y/d ] 

 end 

 function Arenstorf_equations(U, t)          
   
     mu = 0.012277471
           
     x, y, vx, vy = U[1], U[2], U[3], U[4]   
 
     D1 =  ( (x+mu)^2     + y^2 )^1.5
     D2 =  ( (x-(1-mu))^2 + y^2 )^1.5
     
     dxdt = vx  
     dydt = vy  
     dvxdt = x + 2 * vy - (1-mu)*( x + mu )/D1 - mu*(x-(1-mu))/D2
     dvydt = y - 2 * vx - (1-mu) * y/D1 - mu * y/D2
    
     return [ dxdt, dydt, dvxdt, dvydt ] 

 end 

end 

